src/serde_support.rs

```diff
@@ -19,7 +19,8 @@ impl Serialize for Uuid {
         serializer: S,
     ) -> Result<S::Ok, S::Error> {
         if serializer.is_human_readable() {
-            serializer.collect_str(&self.to_hyphenated())
```

Comment:
just a question: was `collect_str` the wrong function to use? curious..

Reply:
I don't think so, it's much more direct and less error prone than manually using the `Display` impls to construct a `str`. This new APIs is only a little bit more complicated to use with `serialize_str` than `collect_str`, so I think it makes sense to switch.